docs/source/user-guide/vision/VisionData.rst

```diff
@@ -0,0 +1,184 @@
+
+    deepchecks_dataset = my_dataset.map(deepchecks_map)
+    vision_data = VisionData(deepchecks_dataset, task_type='classification')
+    vision_data.head()
+
+
+
+Adding Model Predictions
+========================
+Some of deepchecks tests, including the :doc:`model evaluation checks and suite </checks_gallery/vision>`,
+require model predictions in
+order to run. Models prediction are supplied via the batch loader in a similar fashion to the images and labels.
+There are several ways to supply them which can be divide into two categories: Pre-calculated predictions and
+on-demand inference.
+
+On-demand Inference
+-------------------
+In this case we will need to incorporate the model object in the relevant format transformation function. This can be
+done either by using the model as a global variable, creating a wrapper class for the transformation function or
+creating a closure function. We will demonstrate the last option via the pytorch interface.
+
+.. code-block:: python
+
+    import torch
+    from torch.utils.data import DataLoader
+    from deepchecks.vision import VisionData, BatchOutputFormat
+
+    def create_deepchecks_collate(model):
+        def deepchecks_collate(data) -> BatchOutputFormat:
+            # Extracting images and label and predicting using the model
+            raw_images = torch.stack([x[0] for x in data])
+            predictions = model(images)
+            labels = [x[1] for x in data]
+            # Convert ImageNet format images into to [0, 255] range format images.
+            mean, std  = [0.485, 0.456, 0.406], [0.229, 0.224, 0.225]
+            images = np.clip(std * raw_images.permute(0, 2, 3, 1).numpy() + mean, 0, 1) * 255
+            return BatchOutputFormat(images= images, labels= labels, predictions= predictions)
+        return deepchecks_collate
+
+    data_loader = DataLoader(my_dataset, batch_size=64, generator=torch.Generator(),
+                             collate_fn=create_deepchecks_collate(my_model))
+    vision_data = VisionData(data_loader, task_type='classification')
+    vision_data.head()
+
+.. _vision_data__precalculated_predictions:
+Pre-calculated Predictions
+--------------------------
+It is specifically recommended to use this option if your model object is unavailable locally (for example if
+placed on a separate prediction server) or if the predicting process is computationally expensive or time consuming.
+
+In the example below we will read the pre-calculated predictions, as well as the images and labels, from
+a csv file containing the path to the image, the label and the prediction probabilities per sample.
+
+.. code-block:: python
+
+    from PIL import Image
+    from deepchecks.vision import VisionData, BatchOutputFormat
+
+    def data_from_file_generator(batch_size = 64):
+        # Shuffling is a must for generic generators in order to achieve accurate results
+        data = pd.read_csv('classification_data.csv', index_col=0).sample(frac=1)
```

Comment:
Wondering if regular vision users would understand that the small ".sample(frac=1)" is the shuffling. Maybe we should consider doing this in a separate row?